manifests/windows/install.pp

```diff
@@ -45,15 +44,15 @@
   }
 
   $_timestamp = strftime('%Y_%m_%d-%H_%M')
-  $_logfile = "${::env_temp_variable}\\puppet-${_timestamp}-installer.log"
+  $_logfile = windows_native_path("${::env_temp_variable}/puppet-${_timestamp}-installer.log")
   notice ("Puppet upgrade log file at ${_logfile}")
   debug ("Installing puppet from ${_msi_location}")
-  file { "${::env_temp_variable}\\install_puppet.bat":
+  file { "${_installbat}":
     ensure  => file,
     content => template('puppet_agent/install_puppet.bat.erb')
   }->
   exec { 'install_puppet.bat':
-    command => "${::system32}\\cmd.exe /c start /b ${_cmd_location} /c \"${::env_temp_variable}\\install_puppet.bat\" ${::puppet_agent_pid}",
+    command => "${::system32}\\cmd.exe /c start /b ${_cmd_location} /c \"${_installbat}\" ${::puppet_agent_pid}",
```

Comment:
I think this is more complicated than necessary - in Beaker, we just shell to {{cmd.exe}} and pass it the batch file.  I also don't see anything being done with the exit codes - `0`, `1641` and `3010` are all success.

https://github.com/puppetlabs/beaker/blob/f3d87f2b3463c2c49e1dd2b3e34b028336e7151e/lib/beaker/dsl/install_utils/windows_utils.rb#L120-L127